docs/notebooks/feasible_sets.pct.py

```diff
@@ -0,0 +1,224 @@
+#
+
+# %%
+# %matplotlib inline
+
+# silence TF warnings and info messages, only print errors
+# https://stackoverflow.com/questions/35911252/disable-tensorflow-debugging-information
+import os
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+
+import tensorflow as tf
+tf.get_logger().setLevel("ERROR")
+
+import numpy as np
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+
+# %% [markdown]
+# ## A toy problem
+#
+# Throughout the tutorial we will use the standard Branin function as a stand-in for an expensive-to-evaluate system. We create a failure region by thresholding the value at 80, space with value above 80 is considered a failure region. This region needs to be learned as efficiently as possible by the active learning algorithm.
+#
+# Note that if we are interested in a feasibility region instead, it is simply a complement of the failure region, space with the value below 80.
+#
+# We illustrate the thresholded Branin function below, you can note that above the threshold of 80 there are no more values observed.
+
+# %%
+from trieste.objectives import branin, BRANIN_SEARCH_SPACE
+from util.plotting_plotly import plot_function_plotly
+from trieste.space import Box
+
+search_space = BRANIN_SEARCH_SPACE
+
+# threshold is arbitrary, but has to be within the range of the function
+threshold = 80
+
+# define a modified branin function
+def thresholded_branin(x):
+    y = np.array(branin(x))
+    y[y > threshold] = np.nan
+    return tf.convert_to_tensor(y.reshape(-1, 1), x.dtype)
+
+# illustrate the thresholded branin function
+fig = plot_function_plotly(
+    thresholded_branin, search_space.lower, search_space.upper, grid_density=700
+)
+fig.update_layout(height=800, width=800)
+fig.show()
+
+
+# %% [markdown]
+# We start with a small initial dataset where our expensive-to-evaluate function is evaluated on points coming from a space-filling Halton sequence.
+
+# %%
+import trieste
+
+observer = trieste.objectives.utils.mk_observer(branin)
+
+num_initial_points = 12
+initial_query_points = search_space.sample_halton(num_initial_points)
+initial_data = observer(initial_query_points)
+
+
+# %% [markdown]
+# ## Probabilistic model of the objective function
+#
+# Just like in sequential optimization, we use a probabilistic model of the objective function. Acquisition functions will exploit the predictive posterior of the model to identify the failure region. We use a `GPR` model from the GPflow library to formulate a Gaussian process model, wrapping it in Trieste's `GaussianProcessRegression` model wrapper.
+
+# %%
+import gpflow
+from trieste.models.gpflow.models import GaussianProcessRegression
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern32(variance=variance, lengthscales=[2, 2])
```

Comment:
lengthscales seem really high... change to 0.2? possibly, also add the priors we used in other notebooks because this is good practice?

Reply:
this is unscaled branin version though?

Reply:
BRANIN_SEARCH_SPACE = Box([0.0], [1.0]) ** 2 so 0.2 is a much more relevant value than 2. variance is based on data so this one is correct

Reply:
added priors, all works pretty much the same